jobbergateapi2/tests/apps/users/test_routers.py

```diff
@@ -204,3 +204,95 @@ async def test_users_me(client, user_data):
 
     data = response.json()
     assert data["email"] == user_data["email"]
+
+
+@pytest.mark.asyncio
+@database.transaction(force_rollback=True)
+async def test_update_user(client, user_data):
+    """
+    Test update an User.
+    """
+    user = [UserCreate(id=1, **user_data)]
+    await insert_objects(user, users_table)
+
+    response = client.put("/users/1", json={"full_name": "new name"})
+
+    assert response.status_code == status.HTTP_201_CREATED
+    data = response.json()
+
+    assert data["full_name"] == "new name"
+    assert data["id"] == 1
+
+    query = users_table.select(users_table.c.id == 1)
+    user = User.parse_obj(await database.fetch_one(query))
+
+    assert user is not None
+    assert user.full_name == "new name"
+
+
+@pytest.mark.asyncio
+@database.transaction(force_rollback=True)
+async def test_update_user_superuser(client, user_data):
+    """
+    Test update an User to change the superuser status.
+    """
+    user = [UserCreate(id=1, is_superuser=True, **user_data)]
+    await insert_objects(user, users_table)
+
+    response = client.put("/users/1", json={"is_superuser": False})
+
+    assert response.status_code == status.HTTP_201_CREATED
+    data = response.json()
+
+    assert data["is_superuser"] is False
+    assert data["id"] == 1
+
+    query = users_table.select(users_table.c.id == 1)
+    user = User.parse_obj(await database.fetch_one(query))
+
+    assert user is not None
```

Comment:
is this redundant if we test is_superuser below?

Reply:
It is not, one is checking the return of the request, and the other is checking the data in the database